aiidalab_widgets_base/computational_resources.py

```diff
@@ -129,16 +139,13 @@ def __init__(self, description="Select code:", path_to_root="../", **kwargs):
         self.refresh()
 
     def quick_setup(self, _=None):
-        def setup_code():
-            self.aiida_code_setup.computer.refresh()
-            self.aiida_code_setup.computer.value = self.aiida_computer_setup.label.value
-            self.aiida_code_setup.on_setup_code(on_success=self.refresh)
-
-        def setup_code_and_computer():
-            self.aiida_computer_setup.on_setup_computer(on_success=setup_code)
-
+        """Go through all the setup steps automatically."""
         with self.hold_trait_notifications():
-            self.ssh_computer_setup.on_setup_ssh(on_success=setup_code_and_computer)
+            success = self.ssh_computer_setup.on_setup_ssh()
+            if success:
+                success = self.aiida_computer_setup.on_setup_computer()
+            if success:
+                self.aiida_code_setup.on_setup_code()
```

Comment:
```suggestion
            if success:
                if self.aiida_computer_setup.on_setup_computer():
                    self.aiida_code_setup.on_setup_code()
```
However, I think it would be better to use an exception to indicate a failure state.